qiskit_experiments/test/iq_pulse_backend.py

```diff
@@ -0,0 +1,444 @@
+
+    @staticmethod
+    def _get_info(instruction: CircuitInstruction) -> Tuple[Tuple[int], Tuple[float], str]:
+        p_dict = instruction.operation
+        qubit = tuple(int(str(val)[-2]) for val in instruction.qubits)
+        params = tuple(float(val) for val in p_dict.params)
+        return qubit, params, p_dict.name
+
+    def iq_data(
+        self,
+        probability: np.ndarray,
+        shots: int,
+        centers: List[Tuple[float, float]],
+        width: float,
+        phase: Optional[float] = None,
+    ) -> Tuple[List, List]:
+        """Generates IQ data for each logical levels
+
+        Parameters
+        ----------
+        probability : np.ndarray
+            probability of occupation
+        shots : int
+            Number of shots
+        centers : List[Tuple[float, float]]
+            The central i and q points for each level
+        width : float
+            Width of IQ data distribution
+        phase : float, optional
+            Phase of IQ data, by default 0
+
+        Returns
+        -------
+        Tuple[List,List]
+            (I,Q) data
+        """
+        counts_n = np.random.multinomial(shots, probability / sum(probability), size=1).T
+
+        full_i = []
+        full_q = []
+
+        for idx, count_i in enumerate(counts_n):
+            full_i.append(np.random.normal(loc=centers[idx][0], scale=width, size=count_i))
+            full_q.append(np.random.normal(loc=centers[idx][1], scale=width, size=count_i))
+
+        full_i = list(chain.from_iterable(full_i))
+        full_q = list(chain.from_iterable(full_q))
+
+        if phase is not None:
+            complex_iq = (full_i + 1.0j * full_q) * np.exp(1.0j * phase)
+            full_i, full_q = complex_iq.real, complex_iq.imag
+
+        full_iq = 1e16 * np.array([[full_i], [full_q]]).T
+        return full_iq.tolist()
+
+    def _state_to_measurement_data(
+        self,
+        state: np.ndarray,
+        shots: int,
+        meas_level: MeasLevel,
+        meas_return: MeasReturnType,
+    ) -> Union[Dict[str, int], complex]:
+        """Convert the state vector to IQ data or counts."""
+        if self.noise is True:
+            state = state.reshape(self.logical_levels, self.logical_levels)
+            state = DensityMatrix(state / np.trace(state))
+        else:
+            state = Statevector(state / np.linalg.norm(state))
+
+        if meas_level == MeasLevel.CLASSIFIED:
+            measurement_data = state.sample_counts(shots)
+
+        elif meas_level == MeasLevel.KERNELED:
+            # TODO: don't hardcode number of levels:
+            # a) move centers infor to subclass OR
+            # b) take system dims parameter
+            measurement_data = self.iq_data(
+                state.probabilities(), shots, [(-1, -1), (1, 1), (0, np.sqrt(2))], 0.2
+            )
+            if meas_return == "avg":
+                measurement_data = np.average(np.array(measurement_data), axis=0)
```

Comment:
Can we have an extra function that generates the centers?
```suggestion
            centers = self.iq_cluster_centers(kwargs)
            measurement_data = self.iq_data(state.probabilities(), shots, centers, 0.2)
            if meas_return == "avg":
                measurement_data = np.average(np.array(measurement_data), axis=0)
```
Then add the following:
```python
def iq_cluster_centers(self, **kwargs) -> List[Tuple[int, int]]:
    """A function to provide the points for the IQ centers when doing readout.
    
    Subclasses can override this function, for instance, to provide circuit dependent
    IQ cluster centers. If this function is not overridden then the IQ cluster centers returned
    are evenly distributed on the unit sphere in the IQ plane with |0> located at IQ point (0, 1).
    """
    theta = 2 * np.pi / self.model_dim
    return [(np.cos(idx * theta), np.sin(idx * theta) for idx in range(self.model_dim)]
```